.github/workflows/spelling.yml

```diff
@@ -7,24 +41,82 @@ on:
 jobs:
   spelling:
     name: Spell checking
+    permissions:
+      contents: read
+      pull-requests: read
+      actions: read
+      security-events: write
```

Comment:
This is future-proofing. You can remove/ignore it at this time.

At a future point in time, I expect to switch to using `sarif` reporting instead of the ::error/comment model I'm using now. But it's very much a WIP (I used it while developing this PR), and I'm not sure it'll make the next release.